codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -40,11 +40,14 @@ private val suffixSequence = sequenceOf("") + generateSequence(2) { it + 1 }.map
  * @param ctx The surrounding [CodegenContext].
  * @param writer The [KotlinWriter] to generate code into.
  * @param shape The modeled [Shape] on which this JMESPath expression is operating.
+ * @param topLevelParentName The name used to reference the top level "parent" of an expression during codegen.
+ * Defaults to `it`. E.g. `it.field`. Won't be used otherwise.
```

Comment:
> Won't be used otherwise.

nit: This is confusing to me, is it necessary to say this?

Reply:
It's not necessary, I see how it's confusing